### ISSUE TITLE
Avoid settings reset after Windows build updates

### ISSUE THREAD

**Issue by valinet** [valinet/ExplorerPatcher#576]
### Discussed in https://github.com/valinet/ExplorerPatcher/discussions/536

<div type='discussions-op-text'>

<sup>Originally posted by **septsea** December  9, 2021</sup>
I am using Windows 11 Dev. I updated my OS today, and I found that some ExplorerPatcher settings had been reset, including:
- Automatically hide the taskbar
- Hide Control Center button
- "When clicking a system icon in the tray, open:" settings
- Maximum number of frequent apps to show
- Remember last used section in this window
- Receive pre-release versions, if available

Perhaps there are more.

Not all the settings are reset, though. For example, I prefer a small taskbar, and the height is not changed after a system update.

The issue can be reproduced. I updated the OS of one of my virtual machines, and the same issue occurred.
</div>

**Comment by valinet** [comment]
This has to be fixed at some point, I believe. It's not that big of a deal, but annoying.

EP stores its settings in `HKCU\Software\Microsoft\Windows\CurrentVersion\Explorer\ExplorerPatcher`. On some Windows build updates (**NOT** CU updates, rather Insider Preview updates, for now), it seems that Windows deletes all foreign (unrecognized) keys in the parent `Explorer` subkey because reasons.

I know, the settings *shouldn't* be really stored there. I did it originally because that's one of the few places in registry that the UWP `StartMenuExperienceHost.exe` can also access and never knew this was a thing.

The solution should do things like:

* Store settings in traditional places, like `HKCU\Software\ExplorerPatcher`, for example.
* Find some some other place that `StartMenuExperienceHost.exe` can read from and store Start menu related settings there. Ideally, it would read from that same path, but I am not sure about it. Have to investigate.
* Modify places in code where the old path is refrenced to the new path (there is mostly a `#define` that controls this, so it should be pain free to do, but ofc, testing is required)
* Modify the UI file/settings backend file (`settings.reg`) and change all references to the old path to the new path.
* Provide a one-time migration mehanism that would automagically work in the background when the next update is seeded and that would migrate existing settings from old path to the new path.

**Comment by radixs** [comment]
Confirmed, I have experienced a missing clock/network etc. icons that were previously configured. I had to re-enable them.

Also when I click the gear icon (settings icon) and I get the settings menu widget pop-up - it is cropped at the top (My task bar is set to top).

Strangely, as it all happened yesterday - it seems to have fixed itself (though without re-enabling the icons) and the widget was working fine, I also got the clock along with date (!) back. I have shut down the computer and today it is back to being cropped and the icons missing.
